From d7a95045066f1fdec5107eb314e7a97ea9575113 Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Wed, 12 Aug 2026 15:58:02 +0300
Subject: [PATCH] board: solidrun: lx2160acex7: remove superfluous hiding
 mc-mem

Drop code that hides mc-mem by cutting an artificial hole into dram
address space. This is already handled properly by adding a reserved-mem
node to device-tree by fdt_reserve_mc_mem.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 board/solidrun/lx2160acex7/lx2160a.c | 49 ----------------------------
 1 file changed, 49 deletions(-)

diff --git a/board/solidrun/lx2160acex7/lx2160a.c b/board/solidrun/lx2160acex7/lx2160a.c
index bdde9d1880a..ed8446c3d1a 100644
--- a/board/solidrun/lx2160acex7/lx2160a.c
+++ b/board/solidrun/lx2160acex7/lx2160a.c
@@ -318,14 +318,6 @@ void board_setup_fdt_serdes_ports(void *fdt);
 
 int ft_board_setup(void *blob, struct bd_info *bd)
 {
-	int i;
-	u16 mc_memory_bank = 0;
-
-	u64 *base;
-	u64 *size;
-	u64 mc_memory_base = 0;
-	u64 mc_memory_size = 0;
-	u16 total_memory_banks;
 	int err;
 
 	err = fdt_increase_size(blob, 512);
@@ -337,47 +329,6 @@ int ft_board_setup(void *blob, struct bd_info *bd)
 
 	ft_cpu_setup(blob, bd);
 
-	fdt_fixup_mc_ddr(&mc_memory_base, &mc_memory_size);
-
-	if (mc_memory_base != 0)
-		mc_memory_bank++;
-
-	total_memory_banks = CONFIG_NR_DRAM_BANKS + mc_memory_bank;
-
-	base = calloc(total_memory_banks, sizeof(u64));
-	size = calloc(total_memory_banks, sizeof(u64));
-
-	/* fixup DT for the three GPP DDR banks */
-	for (i = 0; i < CONFIG_NR_DRAM_BANKS; i++) {
-		base[i] = gd->bd->bi_dram[i].start;
-		size[i] = gd->bd->bi_dram[i].size;
-	}
-
-#ifdef CONFIG_RESV_RAM
-	/* reduce size if reserved memory is within this bank */
-	if (gd->arch.resv_ram >= base[0] &&
-	    gd->arch.resv_ram < base[0] + size[0])
-		size[0] = gd->arch.resv_ram - base[0];
-	else if (gd->arch.resv_ram >= base[1] &&
-		 gd->arch.resv_ram < base[1] + size[1])
-		size[1] = gd->arch.resv_ram - base[1];
-	else if (gd->arch.resv_ram >= base[2] &&
-		 gd->arch.resv_ram < base[2] + size[2])
-		size[2] = gd->arch.resv_ram - base[2];
-#endif
-
-	if (mc_memory_base != 0) {
-		for (i = 0; i <= total_memory_banks; i++) {
-			if (base[i] == 0 && size[i] == 0) {
-				base[i] = mc_memory_base;
-				size[i] = mc_memory_size;
-				break;
-			}
-		}
-	}
-
-	fdt_fixup_memory_banks(blob, base, size, total_memory_banks);
-
 #ifdef CONFIG_USB_HOST
 	fsl_fdt_fixup_dr_usb(blob, bd);
 #endif
-- 
2.43.0

